tools/validator.py

```diff
@@ -52,6 +52,86 @@ def _CheckRegistryKeyPath(self, filename, artifact_definition, key_path):
 
     return result
 
+  def _CheckWindowsPath(self, filename, artifact_definition, source, path):
+    """Checks if a path is a valid Windows path.
+
+    Args:
+      filename (str): name of the artifacts definition file.
+      artifact_definition (ArtifactDefinition): artifact definition.
+      source (SourceType): source definition.
+      path (str): path to validate.
+
+    Returns:
+      bool: True if the Windows path is valid.
+    """
+    result = True
+
+    number_of_forward_slashes = path.count('/')
+    number_of_backslashes = path.count('\\')
+    if (number_of_forward_slashes < number_of_backslashes and
+        source.separator != '\\'):
+      logging.warning((
+          'Incorrect path separator: {0:s} in path: {1:s} defined '
+          'by artifact definition: {2:s} in file: {3:s}').format(
+              source.separator, path, artifact_definition.name,
+              filename))
+      result = False
+
+    if source.separator == '\\':
+      path_lower = path.lower()
+      path_segments = path_lower.split(source.separator)
+
+      if path_segments[0].startswith('%%users.') and path_segments[0] not in (
```

Comment:
Will this check work for this path I found in the same pr?
'%%users.localappdata%%\ConnectedDevicesPlatform\L.%%users.username%%\ActivitiesCache.db'

In GRR we use a regex to get the interpolations our, have a look at https://github.com/google/grr/blob/8123918157e0c53a2ba36d225cec542c6ee4524d/grr/core/grr_response_core/lib/artifact_utils.py#L75 Maybe you can reuse that code?

Reply:
Good point tracking this in https://github.com/ForensicArtifacts/artifacts/issues/310 for a follow up CL